File: Python/0771_jewels_and_stones/solution.py
def jewels_and_stones(jewels,stones):
    jewel_dict = {}
    for letter in stones:
        if letter in jewels:
            if letter not in jewel_dict:
                jewel_dict[letter] = 1
            else:
                jewel_dict[letter] += 1
    return sum(jewel_dict.values())     

jewels = "aA"
stones = "aAAbbbb"
print(jewels_and_stones(jewels,stones))

File: JavaScript/0771_jewels_and_stones/solution.js
function jewels_and_stones(jewels,stones) {
    let jewel_dict = {};
    for (let letter of stones) {
        if (jewels.includes(letter)) {
            if (letter in jewel_dict) {
                jewel_dict[letter] ++;
            } else {
                jewel_dict[letter] = 1;
            }
        }
    }
    const sum = Object.values(jewel_dict).reduce((acc, val) => acc + val, 0);
    return sum;
}

const jewels = "aA";
const stones = "aAAbbbb";
console.log(jewels_and_stones(jewels,stones));